modules/custom/az_migration/src/Plugin/migrate/source/AZNodeWithFieldCollection.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Drupal\az_migration\Plugin\migrate\source;
+
+use Drupal\migrate\Row;
+use Drupal\node\Plugin\migrate\source\d7\Node as D7Node;
+
+/**
+ * Extends D7Node source plugin to convert field collection content into paragraph content.
+ *
+ * Available configuration keys:
+ * - node_type: The node_types to get from the source - can be a string or
+ *   an array. If not declared then nodes of all types will be retrieved.
+ * - field_collection_names: The field_collection types to get from the source - can be a string or
+ *   an array. If not declared then nothing will be retrieved.
+ *
+ * Examples:
+ *
+ * @code
+ * source:
+ *   plugin: az_node_with_field_collection
+ *   node_type: page
+ *   field_collection_names: field_accordion
+ * @endcode
+ *
+ * In this example nodes of type page, and the accordion field collection are retrieved from the source database
+ *
+ * @code
+ * source:
+ *   plugin: az_node_with_field_collection
+ *   node_type: [page, test]
+ *   field_collection_names:
+ * .    - field_accordion
+ *     - field_contacts
+ * @endcode
+ *
+ * In this example nodes of type page and test along with the field_accordion and
+ * field_contacts field collections are retrieved from the source database.
+ *
+ * For additional configuration keys, refer to the parent classes.
+ *
+ * @see \Drupal\migrate\Plugin\migrate\source\SqlBase
+ * @see \Drupal\migrate\Plugin\migrate\source\SourcePluginBase
+ * @see \Drupal\node\Plugin\migrate\source\d7\Node
+ *
+ * @MigrateSource(
+ *   id = "az_node_with_field_collection",
+ *   source_module = "az_migration"
+ * )
+ */
+class AZNodeWithFieldCollection extends D7Node {
```

Comment:
Nitpick: we discussed whether the `AZ` prefixes are needed or not in our Wednesday workshop.  I'm thinking that since the PHP class is already namespaced with the module name (via the `namespace` line), the only thing that really _needs_ the `AZ` prefix is probably the machine name / ID for the plugin (which yours already has).

Having AZ here doesn't really break anything and our existing plugins are not consistent so feel free to ignore this one if you want.

`az_` prefixed machine names / IDs and non-prefixed class names seems like a good convention to adopt going forward, though, IMHO.

```suggestion
class NodeWithFieldCollection extends D7Node {
```

Reply:
Agreed

Reply:
I'm going to bring this in, but in general, it seems like some of these plugins are really useful outside of just QS1 -> QS2 and could omit the AZ prefix; but it makes sense to me to keep it for stuff that is actually really specifically for QS content.

Reply:
FWIW, the purpose of the `az_` / `AZ` prefixes (in theory) has always just been to avoid naming conflicts with other modules / site-specific configurations / etc.